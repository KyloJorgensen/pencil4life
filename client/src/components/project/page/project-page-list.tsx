'use strict';

import * as React from 'react';
import { Link } from 'react-router-dom';
import ProjectPageListItem from './project-page-list-item';
import { IUserContext, userConsumer } from '../../user/user-provider';
import { IProjectPageContext, projectPageConsumer, getProjectPagesQuery } from './project-page-provider';


export interface ProjectPageListProps {
	projectPage: IProjectPageContext;
	user: IUserContext;
	limit: number;
	discontinued: boolean;
	displayDetails?: boolean;
}

class ProjectPageList extends React.Component<ProjectPageListProps> {
	public static defaultProps = {
		displayDetails: true,
	};
    constructor(props) {
        super(props);
    }

    componentDidMount() {
    	let query: getProjectPagesQuery = {};
    	if (this.props.limit) {
    		query.limit = this.props.limit;
    	}
		this.props.projectPage.getProjectPages(this.props.projectPage._projectId, query)
    }

    componentDidUpdate(prevProps, prevState, snapshot) {
    	let update = false;
    	let prevDiscontinued = prevProps.discontinued || false;
    	let currentDiscontinued = this.props.discontinued || false;

    	if (prevDiscontinued != currentDiscontinued) {
	    	update = true;
    	}

    	if (update) {
    		let query = {
    			discontinued: this.props.discontinued || false,
    			limit: this.props.limit || 10,
    		}
    		this.props.projectPage.getProjectPages(this.props.projectPage._projectId, query);
    	}

    }

	render() {
		const { displayDetails } = this.props;
		const { projectPages, _projectId } = this.props.projectPage;

		let ProjectPageList = [];
		if (projectPages) {
            projectPages.forEach((_projectPageId) => {
				ProjectPageList.push(<ProjectPageListItem key={_projectPageId} _projectPageId={_projectPageId} displayDetails={displayDetails} />)
			});
		}
		return (
			<div className="project-page-list-wrapper" >
				{this.props.user.admin ? (
					<div>
						<Link to={"/project/new/"+_projectId}>NEW PAGE</Link>
					</div>
				) : ''}
				<ul className='project-list' >
					{ProjectPageList}
				</ul>
			</div>
		);			
	}
};

export default userConsumer(projectPageConsumer(ProjectPageList));
export { ProjectPageList };