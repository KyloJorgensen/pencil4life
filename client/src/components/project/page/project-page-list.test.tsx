'use strict';

import * as React from 'react';
import { shallow } from 'enzyme';
import * as RichTextEditor from 'react-rte'
import { ProjectPageList, ProjectPageListProps } from './project-page-list';

const _defaultProps: ProjectPageListProps = {
    projectPage: {    
        getProjectPages: (params) => {},
        getProjectPage: (params) => {},
        addProjectPage: (params) => {},
        updateProjectPage: (params, callback) => {},
        projectPages: [],
        projectPageList: {},
        page: 0,
        totalPages: 0,
        limit: 0,
        total: 0,
        pageTotal: 0,
        projectPage: {
            _projectPageId: '',
            title: '',
            details: RichTextEditor.createEmptyValue().toString('html'),
            discontinued: false,
            _imageId: '',
            page: 0,
        }
    },
    user: {
        signup: () => {},
          getUser: () => {},
          login: () => {},
          logout: () => {},
          updateUser: () => {},
          updatePassword: () => {},
          email: null,
          firstname: null,
          lastname: null,
          username: null,
          userAccess: false,
          admin: false,
    },
    limit: 20,
    discontinued: false,
};

describe('Render Project Page List', () => {
	it('matches the snapshot', () => {
		let wrapper = shallow(<ProjectPageList {..._defaultProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});