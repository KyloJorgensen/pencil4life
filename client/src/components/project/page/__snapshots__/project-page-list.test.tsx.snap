// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Project Page List matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectPageList
    discontinued={false}
    displayDetails={true}
    limit={20}
    projectPage={
      Object {
        "addProjectPage": [Function],
        "getProjectPage": [Function],
        "getProjectPages": [Function],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "projectPage": Object {
          "_imageId": "",
          "_projectPageId": "",
          "details": "<p><br></p>",
          "discontinued": false,
          "page": 0,
          "title": "",
        },
        "projectPageList": Object {},
        "projectPages": Array [],
        "total": 0,
        "totalPages": 0,
        "updateProjectPage": [Function],
      }
    }
    user={
      Object {
        "admin": false,
        "email": null,
        "firstname": null,
        "getUser": [Function],
        "lastname": null,
        "login": [Function],
        "logout": [Function],
        "signup": [Function],
        "updatePassword": [Function],
        "updateUser": [Function],
        "userAccess": false,
        "username": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <ul
          className="project-list"
        />,
      ],
      "className": "project-page-list-wrapper",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "project-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <ul
            className="project-list"
          />,
        ],
        "className": "project-page-list-wrapper",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "project-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
